src/psyclone/transformations.py

```diff
@@ -2422,37 +2307,27 @@ class GOLoopSwapTrans(Transformation):
      >>> swap = GOLoopSwapTrans()
      >>> new_schedule, memento = swap.apply(schedule.children[0])
      >>> new_schedule.view()
-    '''
 
+    '''
     def __str__(self):
         return "Exchange the order of two nested loops: inner becomes " + \
                "outer and vice versa"
 
-    @property
-    def name(self):
-        '''Returns the name of this transformation as a string.'''
-        return "GOLoopSwap"
-
     def validate(self, node_outer, options=None):
-        # pylint: disable=no-self-use
         '''Checks if the given node contains a valid Fortran structure
         to allow swapping loops. This means the node must represent
         a loop, and it must have exactly one child that is also a loop.
 
         :param node_outer: a Loop node from an AST.
         :type node_outer: py:class:`psyclone.psyir.nodes.Loop`
         :param options: a dictionary with options for transformations.
-        :type options: dictionary of string:values or None
+        :type options: dict of string:values or None
 
         :raises TransformationError: if the supplied node does not\
                                      allow a loop swap to be done.
-         '''
 
-        from psyclone.psyir.nodes import Loop
-        if not isinstance(node_outer, Loop):
-            raise TransformationError("Error in GOLoopSwap transformation. "
-                                      "Given node '{0}' is not a loop."
-                                      .format(node_outer))
+        '''
+        super(GOLoopSwapTrans, self).validate(node_outer, options=options)
```

Comment:
You should check inner here too I think.

Reply:
Done.

Reply:
I don't think 2351 covers the same checks as the base class validate would. The latter checks 1) if it is a loop 2) if the loop is fully formed 3) all children of the loop are valid 4) a NEMO-specific constraint on a where statement. The former checks whether it is a loop (so simply replicates check 1.)

So I still think checking inner would make sense.

Additionally, note that on line 2334, the check is for GOLoop whereas on line 2351 it is for Loop. I think it should be for GOLoop in both cases.

Reply:
Oh yes, those are good points. I've updated the code and added a new test.